src/evaluate/hub.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict
+
+import requests
+from datasets.utils.metadata import known_task_ids
+from huggingface_hub import dataset_info, model_info
+from huggingface_hub.repocard import metadata_update
+
+
+def get_allowed_tasks(tasks_dict):
+    return list(tasks_dict.keys()) + [subtask for task in tasks_dict.values() for subtask in task.get("subtasks", [])]
```

Comment:
As discussed offline this should only use the `tasks_dict.keys()` and not the subtasks. cc@lewtun